lib/matplotlib/collections.py

```diff
@@ -2016,6 +2016,44 @@ def set_paths(self):
             self._meshWidth, self._meshHeight, self._coordinates)
         self.stale = True
 
+    def set_array(self, A):
+        if isinstance(self, QuadMesh) and A is not None:
+            shape = A.shape
+            faulty_data = False
+            width, height = self._meshWidth, self._meshHeight
+
+            if self._shading == 'flat':
+                if len(A.shape) == 1:
```

Comment:
Do we allow `len(A.shape())==1` arrays to be passed and they get reshaped somehow? If so, I'm not sure why we would block other arrays that fit?  I tried to track down what happens but could not quickly...

Reply:
I was under the assumption that we only allow flattened arrays that fit, isn't that true?

Reply:
> I was under the assumption that we only allow flattened arrays that fit, isn't that true?

I don't know.  That is what I am asking.